File: Python/p002.py
#! /usr/bin/env python3
"""Solves problem 002 from the Project Euler website"""

from common.fibonacci import fibonacci_numbers_below


def solve():
    """Solve the problem and return the result"""
    fibs = fibonacci_numbers_below(4000000)

    result = 0

    for x in fibs:
        if x % 2 == 0:
            result += x

    return result


if __name__ == '__main__':
    print(solve())

File: project_euler_rust/src/bin/p002.rs
use project_euler_rust::fibonacci_below;

fn main() {
    let fibonacci_numbers = fibonacci_below(4_000_000);
    let mut result: u64 = 0;

    for number in fibonacci_numbers {
        if number % 2 == 0 {
            result += number;
        }
    }

    println!("Project Euler problem ###");
    println!();
    println!("Solution: {}", result);
}
